azure/converters/publicips.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+// IPTagsToSDK converts a CAPZ IP tag to an Azure SDK IP tag.
+func IPTagsToSDK(ipTags []infrav1.IPTag) *[]network.IPTag {
```

Comment:
Is there a difference between using `*[]network.IPTag` as the return type and returning `&skdIPTags` and using `[]network.IPTag` as the return type and returning `sdkIPTags`? I believe in both cases we can return `nil` when the length is zero.

Reply:
No difference, mostly that the place the function is called in the IP properties expects a `*[]network.IPTag` so this allows us to call it directly https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2419/files#diff-101a2cb7fa509c5208b45ff64bc92af67c470f1a3de730a5a05f60b95b7b1681R95